### PR TITLE
Add additional specs for search_history

### PR DESCRIPTION
**Pull request by kswanson33**
#51

### CODE REVIEW

File: spec/search_history_spec.rb [hunk 1]
```diff
@@ -1,15 +1,49 @@
 require 'spec_helper'
 
 RSpec.describe Kitchen::SearchHistory do
+  context 'init' do
+    it 'creates instance variables' do
+      sh = described_class.new('foo', 'bar')
+      expect(sh.upstream).to eq('foo')
+      expect(sh.latest).to eq('bar')
```
Comment (jpslav) [inline spec/search_history_spec.rb:8]: way to go with the fake arguments! 👏

File: spec/search_history_spec.rb [hunk 2]
```diff
@@ -1,15 +1,49 @@
...
   context 'empty' do
     it 'converts to an empty array' do
       expect(described_class.empty.to_a).to eq []
     end
   end
 
+  context '#add' do
+    it 'adds nil if given nil value' do
+      expect(described_class.empty.add(nil).to_a).to eq [nil]
+    end
```
Comment (jpslav) [inline spec/search_history_spec.rb:21]: Nevermind ~Looking at the `[css_or_xpath].join(', ')` part of `#add` I'm confused why I did that.  You might look through the codebase and see if we ever pass in an array as the `css_or_xpath` -- that's the only reason I can think of that we'd need a `join` but at the same time if `css_or_xpath` is an array, then we have nested arrays which is also confusing.~

Comment (jpslav) [reply spec/search_history_spec.rb:21]: I see that I used `to_a` when testing, but there might be a better way.  Here you could maybe say

```ruby
expect(described_class.empty.add(nil).latest).to be_nil
```

File: spec/search_history_spec.rb [hunk 3]
```diff
@@ -1,15 +1,49 @@
...
+
+    it 'returns array with upstream and latest' do
+      expect(described_class.empty.add('foo').add(["blah","bar"]).add(['baz']).to_a).to eq(['foo','blah, bar','baz'])
+    end
+  end
+
+  context "#empty?" do
+    it 'returns true if empty' do
+      expect(described_class.empty.empty?).to eq(true)
```
Comment (jpslav) [inline spec/search_history_spec.rb:42]:
```ruby
expect(described_class.empty).to be_empty
```

https://relishapp.com/rspec/rspec-expectations/v/3-10/docs/built-in-matchers/predicate-matchers

File: spec/search_history_spec.rb [hunk 4]
```diff
@@ -1,15 +1,49 @@
...
+    end
+
+    it 'returns false if not empty' do
+      expect(described_class.empty.add('foo').empty?). to eq(false)
```
Comment (jpslav) [inline spec/search_history_spec.rb:46]:
```ruby
expect(described_class.empty).not_to be_empty
```

File: spec/search_history_spec.rb [hunk 5]
```diff
@@ -1,15 +1,49 @@
...
     end
   end
+
+  context '#to_a' do
+    it 'returns [latest] if no upstream' do
+      expect(described_class.empty.add('foo').to_a).to eq(['foo'])
+    end
+
+    it 'returns array with upstream and latest' do
+      expect(described_class.empty.add('foo').add(["blah","bar"]).add(['baz']).to_a).to eq(['foo','blah, bar','baz'])
```
Comment (jpslav) [inline spec/search_history_spec.rb:36]: rubocop should be complaining about the double quoted strings, changing them to single quoted.

**Review by jpslav — CHANGES_REQUESTED** [review]
Looking good!

File: spec/search_history_spec.rb (hunk 3 above)
Comment (kswanson33) [reply spec/search_history_spec.rb:42]: Just to check that I'm understanding correctly -- the predicate matcher `be_empty` will call the `#empty?` method defined on `SearchHistory` under the hood?

Comment (jpslav) [reply spec/search_history_spec.rb]: yep. it'll split `be_empty` to get `empty` and add the `?` for you.

**Review by jpslav — APPROVED** [review]

**Pull request merged**